src/unix/linux_like/linux/mod.rs

```diff
@@ -1314,6 +1442,36 @@ cfg_if! {
             }
         }
 
+        #[cfg(libc_union)]
+        impl ::fmt::Debug for tpacket_req_u {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("tpacket_req_u")
+                    .field("req3", unsafe { &self.req3 })
+                    .finish()
+            }
+        }
+
+        #[cfg(libc_union)]
+        impl ::fmt::Debug for tpacket_bd_header_u {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("tpacket_bd_header_u")
+                    .field("bh1", unsafe { &self.bh1 })
+                    .finish()
+            }
+        }
+
+        #[cfg(libc_union)]
+        impl ::fmt::Debug for tpacket_block_desc {
+            fn fmt(&self, f: &mut ::fmt::Formatter) -> ::fmt::Result {
+                f.debug_struct("tpacket_bd_header_u")
+                    .field("version", &self.version)
+                    .field("offset_to_priv", &self.offset_to_priv)
+                    .field("hdr", &self.hdr)
+                    .finish()
+            }
+        }
+
+
```

Comment:
Reading `req3` on `tpacket_req_u` reads potentially uninitialized data if the variant is actually `req`. It is probably best to just drop these `Debug` implementations entirely, considering they will likely be removed anyway in 1.0.